lib/recording/RecordingEntrySession.js

```diff
@@ -358,27 +388,58 @@ class RecordingEntrySession {
                 }
                 return this.updateJson(Mp4FilesUpdated)
             })
-            .then(()=>{
-                return this.updateEntryDuration()
+            .then(()=> {
+                return this.updateEntryDuration(false)
+            })
+            .then(()=> {
+                let currentTime = new Date();
+                return this.checkIfSplitSession(currentTime);
             })
-            .catch((err) =>{
+            .catch((err) => {
                 this.logger.error("Error occurred while trying to add new chunks: %s", ErrorUtils.error2string(err));
                 // Only method that calls this function is linkAndUpdateJson (above), and serializedActionsPromise cannot be rejected.
                 return Q.resolve();
             })
 
     }
 
-    changeRecordedEntry(recordedEntryId){
+    changeRecordedEntry(recordedEntryId) {
+        // Todo: decide to use only one member containg recordedEntryId be it the this.entryObject or the this.recordedEntryId
         this.recordedEntryId = recordedEntryId;
+        this.entryObject.recordedEntryId = recordedEntryId;
         this.recordingSessionId = this.entryId + '-' + recordedEntryId;
-        this.recordingSessionPath  = this.getRecordingSessionPath(recordedEntryId);
+        this.recordingSessionPath = this.getRecordingSessionPath(recordedEntryId);
         this.logger = loggerBase.getLogger('RecordingEntrySession', '[' + this.recordingSessionId + '] ');
         return this.playlistGenerator.setBasePath(this.recordingSessionPath);
     }
 
     getRecordingSessionPath(recordedEntryId) {
         return path.join(recordingRootPath, this.modeDirName, this.entryId.slice(-1), this.entryId, recordedEntryId);
     }
+
+    checkIfSplitSession(currentTime) {
```

Comment:
name changed to checkIfRecordedEntryIdChanged() which is actually what this method do.